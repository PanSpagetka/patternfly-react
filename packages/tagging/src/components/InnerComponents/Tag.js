import * as React from 'react';
import PropTypes from 'prop-types';
import { Label } from 'patternfly-react';
import { OverlayTrigger, Tooltip } from 'patternfly-react';


const tooltip = (text) => (
  <Tooltip id="tooltip">
    {text}
  </Tooltip>
);

const Tag = ({
  onTagDeleteClick, tagCategory, tagValue, truncate,
}) => (
  <li key={tagValue.id} className="tag">
    <OverlayTrigger
      placement="bottom"
      overlay={tooltip(tagValue.description)}>
      <Label
        key={tagValue.id}
        bsStyle="primary"
        onRemoveClick={() => onTagDeleteClick(tagCategory, tagValue)}
        className="tagColor"
      >
        {truncate(tagValue.description)}
      </Label>
    </OverlayTrigger>
  </li>
);

Tag.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagCategory: PropTypes.PropTypes.shape({ id: PropTypes.number, description: PropTypes.string }).isRequired,
  tagValue: PropTypes.object.isRequired,
  truncate: PropTypes.func,
};

export default Tag;
