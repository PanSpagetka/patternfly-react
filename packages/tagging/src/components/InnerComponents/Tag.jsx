import * as React from 'react';
import PropTypes from 'prop-types';
import { Label } from 'patternfly-react';


const Tag = ({
  onTagDeleteClick, tagCategory, tagValue, truncate,
}) => (
  <li key={tagValue.id} className="tag">
    <Label
      key={tagValue.id}
      bsStyle="primary"
      onRemoveClick={() => onTagDeleteClick(tagCategory, tagValue)}
      className="tagColor"
      title={tagValue.description}
    >
      {truncate(tagValue.description)}
    </Label>
  </li>
);

Tag.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagCategory: PropTypes.PropTypes.shape({ id: PropTypes.number, description: PropTypes.string }).isRequired,
  tagValue: PropTypes.object.isRequired,
  truncate: PropTypes.func,
};

export default Tag;
