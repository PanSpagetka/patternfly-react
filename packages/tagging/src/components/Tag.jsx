import * as React from 'react';
import PropTypes from 'prop-types';
import { Label } from 'patternfly-react';
import styles from './Tag.scss';


const Tag = ({
  onTagDeleteClick, tagCategory, tagValue,
}) => (
  <li key={tagValue.id} className="tag">
    <Label
      key={tagValue.id}
      bsStyle="primary"
      onRemoveClick={() => onTagDeleteClick(tagCategory, tagValue)}
      style={{ color: styles.defaultColor }}
    >
      {tagValue.description}
    </Label>
  </li>
);

Tag.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagCategory: PropTypes.object.isRequired,
  tagValue: PropTypes.object.isRequired,
};

export default Tag;
