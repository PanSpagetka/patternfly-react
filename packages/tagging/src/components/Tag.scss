@import '~patternfly/dist/sass/patternfly/variables';
:export { defaultColor: $color-pf-white; }

.pf-remove-button {
  color: $color-pf-white;
  vertical-align: middle;
}

.tag {
  font-size: 0.8em;
  display: table-cell !important;
  vertical-align:middle;
  span {
    background-color: $color-pf-blue-400;
  }
}
