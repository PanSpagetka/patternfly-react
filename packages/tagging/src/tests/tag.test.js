import React from 'react';
import Tag from '../components/Tag';
import renderer from 'react-test-renderer';

const tagCategory = { description: 'animal', id: 1 };
const tagValue = { description: 'duck', id: 1 };
function onDelete(x) {
  return x;
}

describe('Tag Component', () => {
  it('match snapshot', () => {
    const component = renderer.create(
      <Tag
        tagCategory={tagCategory}
        tagValue={tagValue}
        onTagDeleteClick={onDelete}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
