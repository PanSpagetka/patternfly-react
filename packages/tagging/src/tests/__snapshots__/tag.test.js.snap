// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component match snapshot 1`] = `
<li
  className="tag"
>
  <span
    className="label label-primary"
    style={
      Object {
        "color": "defaultColor",
      }
    }
  >
    duck
    <a
      className="pf-remove-button"
      href="#"
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
        className="pficon pficon-close"
      />
      <span
        className="sr-only"
      >
        Remove
      </span>
    </a>
  </span>
</li>
`;
